cmd/artifactPrepareVersion.go

```diff
@@ -195,24 +192,32 @@ func versioningTemplate(scheme string) (string, error) {
 	return "", fmt.Errorf("versioning scheme '%v' not supported", scheme)
 }
 
-func calculateNewVersion(versioningTemplate, currentVersion, commitID string, includeCommitID bool, t time.Time) (string, error) {
+func calculateNewVersion(versioningTemplate, currentVersion, commitID string, includeCommitID, shortCommitID, unixTimestamp bool, t time.Time) (string, error) {
 	tmpl, err := template.New("version").Parse(versioningTemplate)
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to create version template: %v", versioningTemplate)
 	}
 
+	timestamp := t.Format("20060102150405")
+	if unixTimestamp {
+		timestamp = fmt.Sprint(t.Unix())
+	}
+
 	buf := new(bytes.Buffer)
 	versionParts := struct {
 		Version   string
 		Timestamp string
 		CommitID  string
 	}{
 		Version:   currentVersion,
-		Timestamp: t.Format("20060102150405"),
+		Timestamp: timestamp,
 	}
 
 	if includeCommitID {
 		versionParts.CommitID = commitID
+		if shortCommitID {
+			versionParts.CommitID = commitID[0:7]
```

Comment:
I would vote for using git to determine the correct short hash as the first 7 digits might not be unique. Also 3-6 digits hashes would be possible that way.

Reply:
I did not want to go down the route calling git again.
I think it is okay to go this route since we always have a timestamp which makes it pretty unique ...
We could also go longer e.g. 10 digits but GitHub uses 7 if I am not mistaken.